### DISCUSSION TITLE
로그인 시 채널 인식 최대 개수 질문

### DISCUSSION THREAD

**Discussion by cjh980402** [storycraft/node-kakao#197] [Q&A] [answered]
v3에서는 방이 아무리 많아도 로그인 시에는 최근 100개 방만 인식된 후, 새로 메시지가 오면 목록에 추가되어 인식이 됐는데
v4에서도 마찬가지인가요?

**Comment by storycraft** [comment, marked as answer]
제가 방이 100개가 넘게 있지가 않아서 첫 부분은 잘 모르겠네요. 단 충분히 긴시간동안 채팅이 오지 않은 비활성 상태인 채팅방은 공식 클라이언트도 채팅 데이터 삭제후 재설치시 뜨지 않는것으로 보아 확인 할 수 있는 방법이 없어보입니다.

**Reply by storycraft** [reply]
비활성 채팅방에서 채팅이 올 경우 v4에서는 제대로 추가되지 않는 것 같습니다.

**Reply by storycraft** [reply]
ref #202

**Reply by cjh980402** [reply]
> 비활성 채팅방에서 채팅이 올 경우 v4에서는 제대로 추가되지 않는 것 같습니다.

비활성 채팅방이나 제가 말한 경우에 속해서 인식이 안된 채팅방에서 새로 오는 채팅이랑 아예 새로들어간 방에서 오는 채팅이랑 경우가 다른가요?

**Reply by storycraft** [reply]
새로 들어간 채팅방의 경우 NEWMEM과 같은 패킷을 통해 새로들어간 채팅방을 알수 있으나, 비활성 채팅방의 경우 MSG 패킷등을 통해 채팅이 오는 경우를 제외하면 알 수 있는 방법이 없기에 다릅니다.

**Reply by cjh980402** [reply]
아 답변 감사합니다. 그래도 새 메시지가 오면 MSG 패킷이 오게 되고 채널 목록에 추가가 되는거군요.
그렇다면 위에 경우에서 채널목록에 추가하기 위한 과정이 있을텐데, 그 과정을 이용해서 인식이 안된 방에 한해 방의 id값 등으로는 수동으로 추가가 안되려나요?
